alibi_detect/utils/pytorch/kernels.py

```diff
@@ -33,7 +33,7 @@ class GaussianRBF(nn.Module):
     def __init__(
         self,
         sigma: Optional[torch.Tensor] = None,
-        init_sigma_fn: Callable = sigma_median,
+        init_sigma_fn: Optional[Callable] = None,
```

Comment:
Not sure why this change is needed, esp. since `None` will lead to `sigma_median`? Is it some issue with the `pydantic` models or serialization?

Would need to document that `None` leads to `sigma_median`.

Same for the `tensorflow` version.

Reply:
Yep as you guessed, it's to make it play nice with `pydantic`, i.e. we can have `init_sigma_fn: Optional[str] = None` in `KernelConfig`, instead of needing to have a validator which defaults to `@alibi_detect.utils.tensorflow.kernels.sigma_median` or the `torch` equivalent, dependent on `backend`. The currently implemented behaviour is similar to how we handle `dataset` in `SpotTheDiff` etc (https://github.com/SeldonIO/alibi-detect/pull/497#discussion_r878198807) i.e. default is `None` and then set to backend-specific object internally.

Reply:
Perhaps an issue to make sure that the defaults are documented in the docstrings in all the places where `None` is going to be replaced by something default in the constructor?